flood_adapt/object_model/hazard/event/event.py

```diff
@@ -99,21 +100,69 @@ def timeseries_shape(
 
     @staticmethod
     def read_csv(csvpath: Union[str, Path]) -> pd.DataFrame:
-        """Read a timeseries file and return a pd.Dataframe.
+        """Read a timeseries file and return a pd.DataFrame.
 
         Parameters
         ----------
         csvpath : Union[str, os.PathLike]
-            path to csv file
+            Path to the CSV file.
 
         Returns
         -------
         pd.DataFrame
-            Dataframe with time as index and waterlevel as first column.
+            Dataframe with time as index and waterlevel as the first column.
         """
-        df = pd.read_csv(csvpath, index_col=0, header=None)
+        num_columns = None
+        has_header = None
+
+        with open(csvpath, "r") as f:
+            try:
+                # read the first 1024 bytes to determine if there is a header
+                has_header = csv.Sniffer().has_header(f.read(1024))
+            except csv.Error:
+                # The file is empty
+                has_header = False
+
+            f.seek(0)
+            reader = csv.reader(f, delimiter=",")
+            num_columns = len(next(reader)) - 1  # subtract 1 for the index column
+
+        if has_header is None:
+            raise ValueError(
+                f"Could not determine if the CSV file has a header: {csvpath}."
+            )
+        header = 0 if has_header else None
+        df = pd.read_csv(csvpath, index_col=0, header=header)
+
         df.index.names = ["time"]
-        df.index = pd.to_datetime(df.index)
+
+        if num_columns is None:
+            raise ValueError(
+                f"Could not read the number of columns in the CSV file: {csvpath}."
+            )
+        df.columns = [f"data_{i}" for i in range(num_columns)]
+
+        SUPPORTED_DATETIME_FORMATS = [
+            "%Y-%m-%d %H:%M:%S",
+            "%Y-%m-%d %H:%M",
+            "%Y-%m-%d",
+            "%Y%m%d %H%M%S",
+            "%Y%m%d %H%M",
+        ]
+        datetime_parsed = False
+        for fmt in SUPPORTED_DATETIME_FORMATS:
+            try:
+                df.index = pd.to_datetime(df.index, format=fmt, errors="raise")
+                datetime_parsed = True
+                break
+            except Exception:
```

Comment:
No bare except please. catch the actual exceptions that `pd.to_datetime` can raise (`ValueError` and `ParseError` in this case)